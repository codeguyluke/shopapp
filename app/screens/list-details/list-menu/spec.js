import React from 'react'
import { shallow } from 'enzyme'
import noop from 'lodash/noop'
import ListMenu, { FADE_DURATION } from './index'

describe('ListMenu', () => {
  const component = (
    <ListMenu
      show
      archived={false}
      onClose={noop}
      onArchivePress={noop}
      onTitleChange={noop}
      listTitle="title"
    />
  )

  it('matches snapshot', () => {
    const wrapper = shallow(component)
    expect(wrapper).toMatchSnapshot()
  })

  it('matches snapshot when archived', () => {
    const wrapper = shallow(React.cloneElement(component, { archived: true }))
    wrapper.setState({ showModal: true })
    expect(wrapper).toMatchSnapshot()
  })

  it('handles title change', () => {
    const onTitleChangeSpy = jest.fn()
    const wrapper = shallow(React.cloneElement(component, { onTitleChange: onTitleChangeSpy }))
    wrapper.setState({ showModal: true })
    
    wrapper.find('FormInput').simulate('changeText', 'test title')
    expect(onTitleChangeSpy).toHaveBeenCalledTimes(1)
    expect(onTitleChangeSpy).toHaveBeenCalledWith('test title')
  })

  it('handles archive press', () => {
    const onArchivePressSpy = jest.fn()
    const wrapper = shallow(React.cloneElement(component, { onArchivePress: onArchivePressSpy }))
    wrapper.setState({ showModal: true })
    
    wrapper.find('Button').at(1).simulate('press')
    expect(onArchivePressSpy).toHaveBeenCalledTimes(1)
  })

  it('toggled showModal state', () => {
    const wrapper = shallow(component)
    expect(wrapper.state().showModal).toBe(false)
    wrapper.instance().toggleModal()
    expect(wrapper.state().showModal).toBe(true)
  })

  it('handles animate out when all refs set', async () => {
    const onCloseSpy = jest.fn()
    const viewRefMock = {
      bounceOutDown: jest.fn(() => Promise.resolve()),
    }
    const blurRefMock = {
      fadeOut: jest.fn(() => Promise.resolve()),
    }

    const wrapper = shallow(React.cloneElement(component, { onClose: onCloseSpy }))
    wrapper.instance().setViewRef(viewRefMock)
    wrapper.instance().setBlurRef(blurRefMock)
    await await wrapper.instance().animateOut()

    expect(viewRefMock.bounceOutDown).toHaveBeenCalledTimes(1)
    expect(blurRefMock.fadeOut).toHaveBeenCalledTimes(1)
    expect(blurRefMock.fadeOut).toHaveBeenCalledWith(FADE_DURATION)
    expect(onCloseSpy).toHaveBeenCalledTimes(1)
  })

  it('handles animate out when view ref not set', async () => {
    const onCloseSpy = jest.fn()
    const viewRefMock = {
      bounceOutDown: jest.fn(() => Promise.resolve()),
    }

    const wrapper = shallow(React.cloneElement(component, { onClose: onCloseSpy }))
    await await wrapper.instance().animateOut()
    expect(viewRefMock.bounceOutDown).not.toHaveBeenCalled()
  })

  it('handles animate out when blur ref not set', async () => {
    const onCloseSpy = jest.fn()
    const viewRefMock = {
      bounceOutDown: jest.fn(() => Promise.resolve()),
    }
    const blurRefMock = {
      fadeOut: jest.fn(() => Promise.resolve()),
    }

    const wrapper = shallow(React.cloneElement(component, { onClose: onCloseSpy }))
    wrapper.instance().setViewRef(viewRefMock)
    await await wrapper.instance().animateOut()

    expect(viewRefMock.bounceOutDown).toHaveBeenCalledTimes(1)
    expect(onCloseSpy).toHaveBeenCalledTimes(1)
    expect(blurRefMock.fadeOut).not.toHaveBeenCalled()
  })

  it('returns null when show is false', () => {
    const wrapper = shallow(React.cloneElement(component, { show: false }))
    expect(wrapper.type()).toEqual(null)
  })
})
