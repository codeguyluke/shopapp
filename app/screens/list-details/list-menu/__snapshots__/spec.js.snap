// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListMenu matches snapshot 1`] = `
<Component
  animationType="none"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <withAnimatable(Component)
    animation="fadeIn"
    delay={0}
    direction="normal"
    duration={200}
    iterationCount={1}
    iterationDelay={0}
    onAnimationBegin={[Function]}
    onAnimationEnd={[Function]}
    onTransitionBegin={[Function]}
    onTransitionEnd={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.6)",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    useNativeDriver={false}
  >
    <TouchableWithoutFeedback
      onPress={[Function]}
    >
      <Component
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </TouchableWithoutFeedback>
  </withAnimatable(Component)>
</Component>
`;

exports[`ListMenu matches snapshot when archived 1`] = `
<Component
  animationType="none"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <withAnimatable(Component)
    animation="fadeIn"
    delay={0}
    direction="normal"
    duration={200}
    iterationCount={1}
    iterationDelay={0}
    onAnimationBegin={[Function]}
    onAnimationEnd={[Function]}
    onTransitionBegin={[Function]}
    onTransitionEnd={[Function]}
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.6)",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    useNativeDriver={false}
  >
    <TouchableWithoutFeedback
      onPress={[Function]}
    >
      <Component
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </TouchableWithoutFeedback>
  </withAnimatable(Component)>
  <withAnimatable(Component)
    animation="bounceInUp"
    delay={0}
    direction="normal"
    iterationCount={1}
    iterationDelay={0}
    onAnimationBegin={[Function]}
    onAnimationEnd={[Function]}
    onTransitionBegin={[Function]}
    onTransitionEnd={[Function]}
    style={
      Object {
        "bottom": 0,
        "height": "80%",
        "left": 16,
        "position": "absolute",
        "right": 16,
      }
    }
    useNativeDriver={false}
  >
    <Button
      buttonStyle={
        Object {
          "alignSelf": "flex-end",
          "backgroundColor": "white",
          "height": 40,
          "marginBottom": 16,
          "padding": 0,
          "paddingLeft": 4,
          "paddingRight": 0,
          "width": 40,
        }
      }
      containerViewStyle={
        Object {
          "marginRight": 0,
        }
      }
      icon={
        Object {
          "color": "darkslategrey",
          "name": "close",
          "size": 32,
        }
      }
      onPress={[Function]}
      rounded={true}
    />
    <KeyboardAwareScrollView
      contentContainerStyle={
        Object {
          "backgroundColor": "white",
          "borderRadius": 4,
          "flex": 1,
          "padding": 24,
        }
      }
      enableAutomaticScroll={true}
      enableResetScrollToCoords={true}
      extraHeight={75}
      extraScrollHeight={0}
      keyboardOpeningTime={250}
      viewIsInsideTabBar={false}
    >
      <Button
        backgroundColor="midnightblue"
        buttonStyle={
          Object {
            "marginTop": 32,
          }
        }
        onPress={[Function]}
        rounded={true}
        textStyle={
          Object {
            "fontFamily": "Hind-Medium",
          }
        }
        title="Unarchive list"
      />
    </KeyboardAwareScrollView>
  </withAnimatable(Component)>
</Component>
`;
